### PR TITLE
feat: introduce `init tip` setting for indexer to skip previous block synchronization

### PR DESCRIPTION
**Pull request by EthanYuan**
### What problem does this PR solve?

Problem Summary:

CKB-indexer supports building indexes only for interested blocks. This can be done by setting `block_filter` before this PR. But the filter can only traverse the block from 0 and for the uninterested block also need to save its `Header(BlockNumber, &'a Byte32, bool)`. 

### What is changed and how it works?

 This pr introduces the `init tip` setting, which can be set directly when the indexer initializes the database for the first time, which brings two benefits:

1. synchronization can be started instantly from any height
2. the previous block does not need to save any data

What's Changed:

- Added commented out configuration item to ckb.toml template (as an example)
- Implement `apply_init_tip` function, which is called in `IndexerService::new`
- Necessary unit tests

### Related changes

This feature will also be used for index-r: 

- #4224  

### Check List 

Tests 

- Unit test
- Manual test
  - general: the indexer can skip the synchronization of the block before the init tip, and the init tip block only saves the block `Header(BlockNumber, &'a Byte32, bool)`
  - tip hash doesn't match, it rolls back to empty
  - when the init tip is set greater than the current node tip, there are no side effects, it waits for the node to synchronize

Side effects

- Performance regression
- Breaking backward compatibility

### Release note 

```release-note
Title Only: Include only the PR title in the release note.
```

### CODE REVIEW

File: resource/ckb.toml [hunk 1]
```diff
@@ -199,3 +199,11 @@ block_uncles_cache_size    = 30
...
 # # Indexing the pending txs in the ckb tx-pool
 # index_tx_pool = false
+# # Customize block filtering rules to index only retained blocks
+# block_filter = "block.header.number.to_uint() >= \"0x0\".to_uint()"
+# # Customize cell filtering rules to index only retained cells
+# cell_filter = "let script = output.type;script!=() && script.code_hash == \"0x00000000000000000000000000000000000000000000000000545950455f4944\""
+# # The initial tip number and hash can be set as the starting height for building indexes in indexer-r. Effective only during the initial index creation.
+# init_tip_number = 10000
+# # The initial tip hash must match the tip number; otherwise, it will result in a rollback to empty.
+# init_tip_hash = "0x8fbd0ec887159d2814cee475911600e3589849670f5ee1ed9798b38fdeef4e44"
```
Comment (quake) [inline resource/ckb.toml:209]: suggest to use one block hash parameter only, indexer may use `get_header` rpc to fetch the block number, and start the indexing service only when corresponding header was found.

Comment (EthanYuan) [reply resource/ckb.toml:209]: Entering a hash manually has the added feature that the block height of the node itself may not have been reached yet and `get_header` may fail.

Comment (EthanYuan) [reply resource/ckb.toml:209]: Entering a number manually considering a situation where the block height of the node itself may not have been reached and get_header may fail.

Comment (quake) [reply resource/ckb.toml:209]: please refer to my previous comment:
> start the indexing service only when corresponding header was found.

Comment (EthanYuan) [reply resource/ckb.toml:209]: Another issue is that if the hash is wrong, then there is no chance of rollback.

Comment (quake) [reply resource/ckb.toml:209]: indexer service won't starting if the hash is wrong.

Comment (EthanYuan) [reply resource/ckb.toml:209]: OK, got it.

**Pull request closed without merging**